File: sorting/InsertionSort.c
#include <stdio.h>


void insertionSort(int a[],int length);
int main()
{
	int a[]={8,4,9,7,6,3,2}
	insertionSort(a,8)
	for(int i =0;i<8;i++){
		printf("a[%d]=%d\n",i,a[i]); 
	} 
	return 0;
}
void insertionSort(int a[],int length);
{
	for (int i=1;i<length;i++)
	{
		int key=a[i];
		int j=i-1;

		while (j>=0 && a[j]>key
				{
				a[j+1]=a[j];
				j=j-1;

	
				}

	a[j+1]=key;
	}


}

File: sorting/InsertionSort.py
def insertionSort(arr):
    for i in range(1,len(arr)):
        j=i
        while arr[j-1]>arr[j] and j>0:
            arr[j-1],arr[j]=arr[j],arr[j-1]
            j-=1
arr=[1,2,5,3,6,9,6,3]
insertionSort(arr)
print(arr)
